### PR TITLE
Copy correct Ztunnel build

### PR DESCRIPTION
**Pull request by Stevenjin8**
**Please provide a description of this PR:**

`bin/build_ztunnel.sh` was copying the debug binary regardless of `BUILD_ZTUNNEL_PROFILE`.

I've tested this with `make build` and `make docker`. Not sure how else I can test this.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: bin/build_ztunnel.sh [hunk 1]
```diff
@@ -107,9 +107,18 @@ function maybe_build_ztunnel() {
...
+  local ZTUNNEL_BIN_PATH
+  if [[ "${BUILD_ZTUNNEL_PROFILE:-dev}" == "dev" ]]; then
+      ZTUNNEL_BIN_PATH=debug
+  else
+      ZTUNNEL_BIN_PATH="${BUILD_ZTUNNEL_PROFILE}"
+  fi
+  ZTUNNEL_BIN_PATH="out/rust/${ZTUNNEL_BIN_PATH}/ztunnel"
+
+  echo "Copying $(pwd)/${ZTUNNEL_BIN_PATH} to ${TARGET_OUT_LINUX}/ztunnel"
   mkdir -p "${ISTIO_ZTUNNEL_LINUX_DEBUG_DIR}"
```
[the remark this answers is not in the document]
Comment (ericvn) [reply bin/build_ztunnel.sh:120]: You want to update this line to allow other profiles as this is specific for debug

**Review by ericvn — DISMISSED** [review]

File: bin/build_ztunnel.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stevenjin8) [reply bin/build_ztunnel.sh:120]: It seems like this line doesn't do anything. `ISTIO_ZTUNNEL_LINUX_DEBUG_DIR` isn't really used anywhere in the script, or the repo at all. When I run `make docker`, it runs with `ISTIO_ZTUNNEL_LINUX_DEBUG_DIR=/home/sj/istio/out/linux_amd64/debug`. But when I look into that directory, it is empty.

I think we should delete the line.

**Comment by Stevenjin8** [comment]
@ericvn I see what you mean. It seems like the directory gets created somewhere in the Makefile. Regardless. It doesn't hurt to have it there. I'll updated so it doesn't have the trailing debug.

File: bin/build_ztunnel.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericvn) [reply bin/build_ztunnel.sh]: The problem might be the next line does a copy of ztunnel one directory up from the debug, `/home/sj/istio/out/linux_amd64` in your case. The mkdir -p will make sure that directory is there.

[the remark this answers is not in the document]
Comment (ericvn) [reply bin/build_ztunnel.sh]: Note the next line is copying to `${TARGET_OUT_LINUX}/ztunnel"`

[the remark this answers is not in the document]
Comment (Stevenjin8) [reply bin/build_ztunnel.sh]: Makes sense. I changed it to just `mkdir -p ${TARGET_OUT_LINUX}`

**Pull request merged**